### DISCUSSION TITLE
"Why does my solver become unresponsive in MATLAB when using the 'optimize' function, but works when I remove some of the constraints?"

### DISCUSSION THREAD

**Discussion by Tuzihao654** [yalmip/YALMIP#1426] [General discussion]
I need to use gurobi to solve an optimization problem with hundreds of thousands of constraints. However, after using the optimize function, matlab has no response, and the command line window does not output any content. Is there a problem when yalmip is converted into gurobi model ?

**Comment by johanlofberg** [comment]
My guess is that YALMIP is stalling converting a large amount of quadratic constraints or something like that. If so, you will have to define using low-level conic operator instead.

Only way to find out is to put a break in callgurobi and step through

**Reply by Tuzihao654** [reply]
I have another question. My model is currently a MILP. I found that sometimes when there are more constraints, the gap decreases faster, and when there are fewer constraints, the BestBD decreases more slowly. Is this reasonable?

**Reply by johanlofberg** [reply]
https://yalmip.github.io/slowmilp

Generally, more constraints is better since that cuts of more possibilities

**Comment by Tuzihao654** [comment]
Thanks, I have one more question. Is there any way to change the upper and lower bounds of a binary variable(binvar)?

**Reply by johanlofberg** [reply]
A binary variable is 0 or 1, so it would be silly to have a model where it was constrained further (sujre, you can add the constraint x<=0 or x>=1, but why would you use a variable if you knew it's value)
